lib/puppet/type/notify.rb

```diff
@@ -6,7 +6,7 @@ module Puppet
   Type.newtype(:notify) do
     @doc = "Sends an arbitrary message to the agent run-time log."
 
-    newproperty(:message) do
+    newproperty(:message, :idempotent => false) do
```

Comment:
So bugs that cause idempotency issues will show as corrective_change's today, this is a known limitation. Also some DSL code can do this also, like services with incorrect status handling and no status => overrides.

The future, would hopefully be able to bring detecting differences between bugs and true corrective_changes, but we were trying to avoid boiling the ocean for now.

If you're interested I have some points on next steps, I've put these forward as phase 2 project ideas basically to the project team.